src/components/accessories/billTable/BillTable.tsx

```diff
@@ -245,23 +248,46 @@ export const BillTable: FC<IBillTableProps> = ({
                     onChange={dateFieldHandleOnChange("toDate")}
                   />
                 </div>
-              </div>
-              <div className="row start-sm center-xs">
-                <div className="fullWidth filterBillForm__item">
-                  <PatientAutocomplete
+                <div className="filterBillForm__item">
+                  <DateField
+                    fieldName="month"
+                    views={["month"]}
+                    fieldValue={formik.values.month}
+                    disableFuture={true}
+                    theme="regular"
+                    format="MMMM"
+                    isValid={isValid("month")}
+                    errorText={getErrorText("month")}
+                    label={t("bill.month")}
+                    onChange={dateFieldHandleOnChange("month")}
+                  />
+                </div>
+                <div className="filterBillForm__item">
+                  <DateField
+                    fieldName="year"
+                    views={["year"]}
+                    fieldValue={formik.values.year}
+                    disableFuture={true}
+                    theme="regular"
+                    format="yyyy"
+                    isValid={isValid("year")}
+                    errorText={getErrorText("year")}
+                    label={t("bill.year")}
+                    onChange={dateFieldHandleOnChange("year")}
+                  />
+                </div>
+                <div className="filterBillForm__item">
+                  <PatientPicker
                     theme={"regular"}
                     fieldName="patientCode"
                     fieldValue={formik.values.patientCode}
                     label={t("bill.patient")}
                     isValid={isValid("patientCode")}
                     errorText={getErrorText("patientCode")}
                     onBlur={onBlurCallback("patientCode")}
-                    freeSolo={true}
                   />
```

Comment:
I think that the UX of this component is not the best: I mean if I blur on the text field nothing happens. I discover that in order to search the patient I must click on the icon inside the text fields, this is not usable for the most of the user.

I suggest to open the modal once the user click (or blur) on the textField in order to be more usable and intuitive.

Then inside the modal, I think that the UI is great but I would change the rappresentation of the list's users: instead of have 3 cards per line I would show those cards once per line (like the patient search). In fact, I think you could reuse the same component.

Reply:
PLus the modal's header should be formatted like this: title align to left (or center) and close button to the right. (UX Trick: the cta action should always align to the right)

Reply:
OK

Reply:
> I think that the UX of this component is not the best: I mean if I blur on the text field nothing happens. I discover that in order to search the patient I must click on the icon inside the text fields, this is not usable for the most of the user.
> 
> I suggest to open the modal once the user click (or blur) on the textField in order to be more usable and intuitive.
> 
> Then inside the modal, I think that the UI is great but I would change the rappresentation of the list's users: instead of have 3 cards per line I would show those cards once per line (like the patient search). In fact, I think you could reuse the same component.

About the patient list in the modal, it is responsive, it goes from 3 to 2 and then 1 according to de device size.  why don't we do like that?

Reply:
The card si small, there are (or there could be) some text that outlines outside the card because is too long to handler. In a mobile first UI if we have some space to use, use it at all.

Reply:
Ok you are right

Reply:
I have done it.

Reply:
Ok, I applied it